lib/mix/lib/mix/rebar.ex

```diff
@@ -38,6 +38,26 @@ defmodule Mix.Rebar do
     global_rebar_cmd(manager) || local_rebar_cmd(manager)
   end
 
+  @doc """
+  Returns the version of the available `rebar` command if any
+  """
+  @rebar_vsn_regex ~r/^## Rebar3 (?<vsn>[\w\.\d]+)$/m
```

Comment:
Prefix isn't always `##`. That is only for the ones that are installed with `rebar3 local install`.

The escript will have `%%`. And I just realized only the non-escript has a space there. escript is ` %%Rebar3 <vsn>`.

Reply:
The escript is `%% Rebar3 version`; the actual escript generator thing inserts the space itself:

```
→ head -n2 rebar3
#!/usr/bin/env escript
%% Rebar3 3.14.0-rc1
``` 
Otherwise it is true that it sometimes is ## and sometimes is %%. Also if you use asdf, the shim spacing is different:

```
$ cat `which rebar3`
#!/usr/bin/env bash
# asdf-plugin: rebar 3.13.2
exec /usr/local/opt/asdf/bin/asdf exec "rebar3" "$@"
```

Also the thing to worry about is the semver-like pre-release flags:

```
%% Rebar3 3.14.0-rc1
```

So if you want to be sure, the regex likely looks like `^(##|%%).*[Rr]ebar3 (?<vsn>[\w\d.-]+)$`

Reply:
👍

Reply:
@ferd Does the asdf shim have one or octothorps? Not sure if you typed that out or if it was copy/paste.

Reply:
I don't think the asdf plugin should affect this though. Doesn't mix always use it's own copy of rebar and not the system one?

Reply:
Ah, sorry for the noise. I now see there's an option to use rebar from an env var.

Reply:
Checked with someone using asdf, mix should not end up referring to a shim, just escript.

Reply:
I am still worried about the implications of this version check. It means folks can be using the same Rebar version but installed in different ways and they won't get the same behaviour. :/ Whatever check we do it is impossible to guarantee it will work on all the different ways rebar can be installed.

Reply:
That is correct. But all of that is done as a nicety so mix doesn't have to call and cache the result of `rebar3 version` which successfully prints the full version since before the first alpha release, and which would be the true safest approach.